### PR TITLE
User profile

### PR DESCRIPTION
**Pull request by dpslwk**
Some basic view and edit pages

Update to the search api to return paginated results

### CODE REVIEW

File: app/HMS/Repositories/Doctrine/DoctrineUserRepository.php [hunk 1]
```diff
@@ -50,9 +51,16 @@ public function findOneByEmail(string $email)
...
      * @param  bool $hasAccount limit to users with associated accounts
-     * @return array
+     * @param bool $paginate
+     * @param int    $perPage
+     * @param string $pageName
+     * @return User[]|array|\Illuminate\Pagination\LengthAwarePaginator
      */
-    public function searchLike(string $searchQuery, ?bool $hasAccount = false)
+    public function searchLike(string $searchQuery,
+        ?bool $hasAccount = false,
```
Comment (geeksareforlife) [inline app/HMS/Repositories/Doctrine/DoctrineUserRepository.php]: Error?

File: app/HMS/Repositories/UserRepository.php [hunk 2]
```diff
@@ -33,9 +34,12 @@ public function findOneByEmail(string $email);
...
      * @param  string $searchQuery
      * @param  bool $hasAccount limit to users with associated accounts
-     * @return array
+     * @param bool $paginate
+     * @param int    $perPage
+     * @param string $pageName
+     * @return User[]|array|\Illuminate\Pagination\LengthAwarePaginator
      */
-    public function searchLike(string $searchQuery, ?bool $hasAccount = false);
+    public function searchLike(string $searchQuery, ?bool $hasAccount = false, bool $paginate = false, $perPage = 15, $pageName = 'page');
```
Comment (geeksareforlife) [inline app/HMS/Repositories/UserRepository.php]: Another one?  Is this new PHP syntax that I don't know about?

Comment (dpslwk) [inline app/HMS/Repositories/UserRepository.php]: yeah the ? is declaring it a nullable type, php 7.1 introduced this, mean the value can be NULL or of the type
and yes into this case thats either Null, True or False

File: resources/views/user/edit.blade.php [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+        </div>
+      </div>
+    </div>
+
+    <p>The unlock text is displayed by gatekeeper when opening a door.</p>
+    <div class="form-container">
+      <div class="row">
+        <label for="unlockText" class="form-label">Unlock Text</label>
+        <div class="form-control">
+          <input id="unlockText" type="date" name="unlockText" value="{{ old('unlockText', $user->getProfile()->getUnlockText()) }}">
```
Comment (geeksareforlife) [inline resources/views/user/edit.blade.php]: Don't think you want this to be a date?

**Pull request merged**